packages/next/src/server/app-render/app-render.tsx

```diff
@@ -908,7 +908,10 @@ async function renderToHTMLOrFlightImpl(
     req.originalRequest.on('end', () => {
       const staticGenStore = ComponentMod.workAsyncStorage.getStore()
       const prerenderStore = prerenderAsyncStorage.getStore()
-      const isPPR = !!prerenderStore?.dynamicTracking?.dynamicAccesses?.length
+      const isPPR =
+        prerenderStore && prerenderStore.type === 'prerender'
+          ? !!prerenderStore.dynamicTracking?.dynamicAccesses?.length
+          : false
```

Comment:
I didn't see the original when this landed but isn't this always undefined? this function is created outside the prerenderAsyncStorage scope and the request's originalRequest events shouldn't be run inside this store unless that was added somewhere else. Can land b/c this is semantically equivalent but I suspect this might not be working as intended

Reply:
It seems fragile regardless but the timing of this end call could be inside some context because it doesn’t necessarily have to cary the context of where the listener is added if it’s not instrumented.